### PR TITLE
Fix for BPM Signal Notifications

### PR DESCRIPTION
**Pull request by rkreich**
Discussed with @tan-tan-kanarek

### CODE REVIEW

File: plugins/event_notification/providers/bpm/lib/model/BusinessProcessSignalNotificationTemplate.php [hunk 1]
```diff
@@ -19,9 +19,44 @@ public function __construct()
...
+		$jobId = null;
+		foreach($caseIds as $caseId)
+		{
+			// when we have $currentCaseId, we should only signal to that case
+			if (!is_null($currentCaseId) && $caseId !== $currentCaseId)
+				continue;
+
+			$currentJobData = clone $jobData;
+			$currentJobData->setCaseId($caseId);
+			$this->dispatchJob($scope, $currentJobData);
```
Comment (MosheMaorKaltura) [inline plugins/event_notification/providers/bpm/lib/model/BusinessProcessSignalNotificationTemplate.php]: Once case found - break the loop.

Comment (rkreich) [inline plugins/event_notification/providers/bpm/lib/model/BusinessProcessSignalNotificationTemplate.php]: @MosheMaorKaltura, the default behavior is actually to signal all cases (if $currentCaseId is null). So we should continue with the loop.
@tan-tan-kanarek, does it make sense to keep the code that will allow signaling all cases? Or should we always try to signal a specific case and throw an exception when case was not found on the event data?

Comment (tan-tan-kanarek) [inline plugins/event_notification/providers/bpm/lib/model/BusinessProcessSignalNotificationTemplate.php]: We might have a future use case to signal all cases, but I don't think it's very important for now as all our signaling will have a case id.

Comment (rkreich) [inline plugins/event_notification/providers/bpm/lib/model/BusinessProcessSignalNotificationTemplate.php]: OK. I'll leave the code as is, we might actually want to manually dispatch the template to signal all cases of a specific process.

**Pull request merged**